contracts/instruments/RibbonThetaVaultYearn.sol

```diff
@@ -0,0 +1,720 @@
+        uint256 share,
+        uint256 fee
+    );
+
+    event OpenShort(
+        address indexed options,
+        uint256 depositAmount,
+        address manager
+    );
+
+    event CloseShort(
+        address indexed options,
+        uint256 withdrawAmount,
+        address manager
+    );
+
+    event WithdrawalFeeSet(uint256 oldFee, uint256 newFee);
+
+    event CapSet(uint256 oldCap, uint256 newCap, address manager);
+
+    event ScheduleWithdraw(address account, uint256 shares);
+
+    event ScheduledWithdrawCompleted(address account, uint256 amount);
+
+    /**
+     * @notice Initializes the contract with immutable variables
+     * @param _asset is the asset used for collateral and premiums
+     * @param _weth is the Wrapped Ether contract
+     * @param _usdc is the USDC contract
+     * @param _yearnRegistry is the registry contract for all yearn vaults
+     * @param _swapContract is the Airswap Swap contract
+     * @param _tokenDecimals is the decimals for the vault shares. Must match the decimals for _asset.
+     * @param _minimumSupply is the minimum supply for the asset balance and the share supply.
+     * @param _isPut is whether this is a put strategy.
+     * It's important to bake the _factory variable into the contract with the constructor
+     * If we do it in the `initialize` function, users get to set the factory variable and
+     * subsequently the adapter, which allows them to make a delegatecall, then selfdestruct the contract.
+     */
+    constructor(
+        address _asset,
+        address _factory,
+        address _weth,
+        address _usdc,
+        address _yearnRegistry,
+        address _swapContract,
+        uint8 _tokenDecimals,
+        uint256 _minimumSupply,
+        bool _isPut
+    ) {
+        require(_asset != address(0), "!_asset");
+        require(_factory != address(0), "!_factory");
+        require(_weth != address(0), "!_weth");
+        require(_usdc != address(0), "!_usdc");
+        require(_yearnRegistry != address(0), "!_yearnRegistry");
+        require(_swapContract != address(0), "!_swapContract");
+        require(_tokenDecimals > 0, "!_tokenDecimals");
+        require(_minimumSupply > 0, "!_minimumSupply");
+
+        IRibbonFactory factoryInstance = IRibbonFactory(_factory);
+
+        address adapterAddr = factoryInstance.getAdapter(_adapterName);
+        require(adapterAddr != address(0), "Adapter not set");
+
+        asset = _isPut ? _usdc : _asset;
+        underlying = _asset;
+        collateralToken = IYearnVault(
+            IYearnRegistry(_yearnRegistry).latestVault(_isPut ? _usdc : _asset)
+        );
+        adapter = IProtocolAdapter(adapterAddr);
+        WETH = _weth;
+        USDC = _usdc;
+        SWAP_CONTRACT = ISwap(_swapContract);
+        _decimals = _tokenDecimals;
+        MINIMUM_SUPPLY = _minimumSupply;
+        isPut = _isPut;
+        YEARN_WITHDRAWAL_BUFFER = 5; // 0.05%
+        YEARN_WITHDRAWAL_SLIPPAGE = 5; // 0.05%
+    }
+
```

Comment:
YEARN_WITHDRAWAL_BUFFER: how much more to withdraw relatively (ex: someone requests 1 eth, we withdraw 1.0005 equivalent of eth from our yvweth holdings).

YEARN_WITHDRAWAL_SLIPPAGE: acceptable slippage when withdrawing. 

buffer should be >= slippage. if slippage acceptable is larger than buffer then it defeats the purpose

Reply:
I only see `YEARN_WITHDRAWAL_SLIPPAGE` being referred in the contract, but not `YEARN_WITHDRAWAL_BUFFER`

Reply:
addressed